lib/Parse/ParseDecl.cpp

```diff
@@ -4154,6 +4154,25 @@ ParserStatus Parser::parseDeclAttribute(
     return makeParserSuccess();
   }
 
+  // Rewrite @_unavailableInEmbedded into @available(*, unavailable) when in
+  // embedded Swift mode, or into nothing when in regular mode.
+  if (DK == DAK_Count && Tok.getText() == "_unavailableInEmbedded") {
+    SourceLoc attrLoc = consumeToken();
+    if (Context.LangOpts.hasFeature(Feature::Embedded)) {
+      StringRef Message = "unavailable in embedded Swift", Renamed;
+      auto attr = new (Context) AvailableAttr(AtLoc, SourceRange(AtLoc, attrLoc),
+                  PlatformKind::none,
+                  Message, Renamed, /*RenameDecl=*/nullptr,
+                  llvm::VersionTuple(), SourceRange(),
+                  llvm::VersionTuple(), SourceRange(),
+                  llvm::VersionTuple(), SourceRange(),
+                  PlatformAgnosticAvailabilityKind::Unavailable,
+                  /*Implicit=*/false, /*IsSPI=*/false);
+      Attributes.add(attr);
+    }
```

Comment:
Does this mean that `@_unavailableInEmbedded` is silently dropped when `Feature::Embedded` is not enabled? I think in that case we should
- either emit an error 
- or fall down to the standard attribute parsing logic, which will consequently complain that no type `_unavailableInEmbedded` is defined.

Reply:
The attribute is meant to behave as a no-op when `Feature::Embedded` is not specified, not be diagnosed.

Reply:
The way I conceptualize it is that it's really a macro that expands to `@available(*, unavailable)` when in embedded mode, and expands to nothing otherwise. It could be nice to actually implement it with macros, but I don't think we support macros that just create attributes yet, right?

Reply:
OK, if that’s the intended behavior, I’m fine with it.